example/Makefile

```diff
@@ -1,8 +1,16 @@
-compile:
+.PHONY: default run build
+
+SRCS := $(wildcard ../*.go)
+
+default: run
+
+x.wasm: main.go $(SRCS)
```

Comment:
The original version of PR 5 added a few nice features for reducing the need for building, however it was missing a few dependencies, which made it a little less convenient for sdk work.   Here, we propose creating a dependency the main file and all go files of the sdk, should cover it... but let me know if I am missing something.

See original idea proposed in PR 5: (https://github.com/blocky/basm-go-sdk/pull/5/files/8acf608334a61975476a529591e76a962079b11f#diff-a7f0749bdafe7d4247c8435bbb139a3ca42c20bf58a96868ac286419afb764d1

Reply:
I wonder if we'd want too add the `check` target and build `x.wasm` using docker, so that we can lock down the version of `tiny.go` being used in the build. 

If you don't want to move in that direction, than maybe just add `jq` as a dependency in the readme

Reply:
@mwittie when you say list jq as a dependency, do you mean something more than this? https://github.com/blocky/basm-go-sdk/blob/0fd8744a2ba5b071fde7ee134e84b7398f064336/example/README.md?plain=1#L10

Reply:
@dlm this covers all the build files as far as I can tell 👍🏼

Reply:
@mwittie for the time being, I would prefer that we just document the dependencies and let the developer manage it as they like.  I added a card for adding better tooling for managing the environment.  Note that while what we do for the examples works pretty well, it doesn't solve the problems that we run into when developing the sdk.  Specifically, the hard part is when we need to build against pre-release versions of delphi.